src/SourceBuild/content/eng/tools/BinaryToolKit/BinaryTool.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Logging;
+
+namespace BinaryToolKit;
+
+public partial class BinaryTool
+{
+
+    // Target directory - where to look for binaries
+    private readonly string TargetDirectory;
+
+    // Allowed binaries file - known binaries in the target directory that should not be removed. Needed for validation and cleaning.
+    private readonly string? AllowedBinariesFile;
+
+    // Disallowed source build binaries file - known binaries in the target directory that are disallowed for source-build. Needed for validation.
+    private readonly string? DisallowedSbBinariesFile;
+
+    // Updated allowed binaries file - updated baseline of allowed binaries
+    private readonly string? UpdatedAllowedBinariesFile;
+
+    // Updated disallowed source build binaries file - updated baseline of disallowed source build binaries
+    private readonly string? UpdatedDisallowedSbBinariesFile;
+
+    // New binaries file - list of new binaries detected in the target directory
+    private readonly string? NewBinariesFile;
+
+    // Mode to run the binary detection in
+    private Mode.ModeOptions ModeOption;
+
+    // Logger
+    private ILogger Log;
+
+    public BinaryTool(string targetDirectory, string outputReportDirectory, string? allowedBinariesFile, string? disallowedSbBinariesFile, Mode.ModeOptions mode)
+    {
+        ModeOption = Mode.GetFullMode(mode);
```

Comment:
I don't like the idea that `ModeOption` automatically can't have 50% of it's values after this line, but the type doesn't communicate that. If you wrote a switch statement in the other partial class files, you would still need to handle all of the cases in the enum including the shorthand. Are the shorthand options for mode necessary or just for convenience? If you decide to keep the shorthand around, would this method be better mapping from one enum type to another? For example - `ModeCliOption` to `BinaryToolMode`. Or something.

Reply:
The shorthand options are just for convenience. I don't mind removing them as it simplifies things quite a bit, otherwise I can do two separate enums. I just felt that having separate enums that shared so many values was a bit odd.